### ISSUE TITLE
Unreported crash in testing

### ISSUE THREAD

**Issue by meyercm** [ThrowTheSwitch/Ceedling#138]
#### TL,DR; certain kinds of test errors don't show in the Test summary.

In my typical development cycle today, this happened:

```
> ceedling test:all
...
<Lots of scrolling, and 233 lines of output omitted>
...
--------------------
OVERALL TEST SUMMARY
--------------------
TESTED:  133
PASSED:  123
FAILED:    0
IGNORED:  10
```

Everything looks great, right?  Except that back a few pages, this happened:

```
Test 'test_<some_module>.c'
--------------------------------
ERROR: Found no file '<some_file>.h' in search paths.
rake aborted!

<Large Stacktrace Omitted>

Tasks: TOP => test:all
(See full trace by running task with --trace)
```

I didn't see the error go by, because of the large volume of output. Thankfully, I know that I should have ~180 tests in my total count.  I've contemplated using a script to handle this for me (My CI server does have this script, and greps my test directory to count how many should be there), but frankly, this is the job of `Ceedling`.

I have seen this behavior occur when a test segfaults as well, and I'm confident there are other ways to reproduce the underlying behavior.

I think adding a line to the `OVERALL TEST SUMMARY` output and updating the failed count is the ideal fix, e.g.

```
--------------------
OVERALL TEST SUMMARY
--------------------
COUNT:  189
TESTED:  133
PASSED:  123
FAILED:    > 1
IGNORED:  10
```

This is related to, but a separate issue from #117; I mention it because the fix for this issue should impact the overall return code as well.

**Comment by mvandervoord** [comment]
Yeah, this has been something we've been wrestling with ever since we rerouted everything to go through the ceedling command. It's something I want to fix very soon. 

Thanks for capturing it and I apologize for the inconvenience. You're absolutely correct that this is well within Ceedling's responsibilities.

**Comment by meyercm** [comment]
No need to apologize, thank you for the great tool.  Even with flaws, this is strictly better than the alternatives.  

I'd like to contribute, but as has been pointed out elsewhere (e.g. #70, #89), the barrier to entry is rather high, and every time I've sat down to parse out the codebase and put together a pull request, I've thrown my hands up in frustration.

Thank you for being so responsive.

**Comment by mkarlesky** [comment]
I am closing the very old issue as both types of test build failures — missing file during build and actual test executable crash — have been properly handled for quite a while now. On the latter (for any who come across this closed issue later), please see [Ceedling’s backtrace feature](https://throwtheswitch.github.io/Ceedling/latest/configuration/reference/project/#use_backtrace).
